addons/sale_margin/models/sale_order.py

```diff
@@ -86,3 +86,10 @@ def _compute_margin(self):
             for order in self:
                 order.margin = mapped_data.get(order.id, 0.0)
                 order.margin_percent = order.amount_untaxed and order.margin/order.amount_untaxed
+
+    @api.returns('self', lambda value: value.id)
+    def copy(self, default=None):
+        """Ensure that the margins in the copied order are up to date"""
+        new_order = super().copy(default=default)
+        new_order.sudo().order_line._compute_purchase_price()
+        return new_order
```

Comment:
Why do you need to force the computation manually here ?
AFAIK, if it's not in the copied create values, it should be automatically computed by the ORM :thinking:

Reply:
I think I'll add a test to be sure :+1:

Reply:
I just added a test. Indeed only the `copy=False` is needed to trigger the recomputation of the margins after the order copy.